### ISSUE TITLE
Debugging connection with Vue.js

### ISSUE THREAD

**Issue by denisvitez** [miguelgrinberg/Flask-SocketIO#1467]
**Your question**
Any ideas on how to debug connection between my `flask-socketio` server and Vue.js client using `vue-socket.io-extended`?Basically I see some activity on my server side when my Vue.js site is open (logs attached) but the client says it's not connected and messages don't arrive. What is weird is that there seems to be so much activity in the log (Client is gone gets printed multiple times).

What should I see in my logs if connection would be succesfull?

For the Vue.js side I implemented it as described in their tutorial.

Regarding the versions:
`flask-socketio -> 5.0.1`
`vue-socket.io-extended -> 4.0.6
`
**Logs**
When my client page is open:
`127.0.0.1 - - [19/Jan/2021 20:24:11] "GET /socket.io/?EIO=4&transport=polling&t=NSS6hz- HTTP/1.1" 200 -
1d9e85b937fb4639858c2f1c0fb9a0a1: Sending packet OPEN data {'sid': '1d9e85b937fb4639858c2f1c0fb9a0a1', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
1d9e85b937fb4639858c2f1c0fb9a0a1: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:16] "GET /socket.io/?EIO=4&transport=polling&t=NSS6jCz HTTP/1.1" 200 -
42e4fad6610c4f419dd40525bc5004b9: Sending packet OPEN data {'sid': '42e4fad6610c4f419dd40525bc5004b9', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
42e4fad6610c4f419dd40525bc5004b9: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:19] "GET /socket.io/?EIO=4&transport=polling&t=NSS6jzz HTTP/1.1" 200 -
3e2d02815cd74b7d8951e16b29d906fb: Sending packet OPEN data {'sid': '3e2d02815cd74b7d8951e16b29d906fb', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
3e2d02815cd74b7d8951e16b29d906fb: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:20] "GET /socket.io/?EIO=4&transport=polling&t=NSS6kDT HTTP/1.1" 200 -
6cfb123997ab4390857a183e4f2acb4c: Sending packet OPEN data {'sid': '6cfb123997ab4390857a183e4f2acb4c', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
6cfb123997ab4390857a183e4f2acb4c: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:22] "GET /socket.io/?EIO=4&transport=polling&t=NSS6krS HTTP/1.1" 200 -
38a0c58cf1aa4f619b24f7cc7d58deae: Sending packet OPEN data {'sid': '38a0c58cf1aa4f619b24f7cc7d58deae', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
38a0c58cf1aa4f619b24f7cc7d58deae: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:24] "GET /socket.io/?EIO=4&transport=polling&t=NSS6lOH HTTP/1.1" 200 -
bb4b0eb38a734da98bc4bdd1795e49c0: Sending packet OPEN data {'sid': 'bb4b0eb38a734da98bc4bdd1795e49c0', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
bb4b0eb38a734da98bc4bdd1795e49c0: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:24:30] "GET /socket.io/?EIO=4&transport=polling&t=NSS6mcg HTTP/1.1" 200 -
`
But sometimes I also get the message about the client gone:
`
c38f8aa427f547f1a46df0ad4b765afc: Sending packet OPEN data {'sid': 'c38f8aa427f547f1a46df0ad4b765afc', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
c38f8aa427f547f1a46df0ad4b765afc: Sending packet MESSAGE data 0
127.0.0.1 - - [19/Jan/2021 20:25:52] "GET /socket.io/?EIO=4&transport=polling&t=NSS74e6 HTTP/1.1" 200 -
77fb54552e804d858152bae977fcf2c7: Client is gone, closing socket
77fb54552e804d858152bae977fcf2c7: Client is gone, closing socket`

When I try to send a message from my server I get something like the following:
`127.0.0.1 - - [19/Jan/2021 20:30:11] "POST /test/socketio HTTP/1.1" 200 -
0a4b514e9a1b4c93a639e06386a1a1c3: Client is gone, closing socket
8b99caa0bc7a408993a1304a50a55310: Client is gone, closing socket
8b99caa0bc7a408993a1304a50a55310: Client is gone, closing socket
b0c4d8d8bb4c4f8ea9b122dbfe2a6fdc: Client is gone, closing socket
b0c4d8d8bb4c4f8ea9b122dbfe2a6fdc: Client is gone, closing socket
1fcb9a6796ff477184eb2acf9ceff577: Client is gone, closing socket
1fcb9a6796ff477184eb2acf9ceff577: Client is gone, closing socket
a2601dffaeaf46009cd0a4b15eb5e638: Client is gone, closing socket
a2601dffaeaf46009cd0a4b15eb5e638: Client is gone, closing socket
37faa2e7557e4b71893410e5226b73f0: Client is gone, closing socket
37faa2e7557e4b71893410e5226b73f0: Client is gone, closing socket
0fcec346a5c940a69076731ba269ee30: Client is gone, closing socket
0fcec346a5c940a69076731ba269ee30: Client is gone, closing socket
c5963ccae9cd4275a108e92143451b28: Sending packet MESSAGE data 2["test",{"Message":"CREATE_BRIDGE_CONFIG"}]
5d3ecd8d163b4eb9bc5cd06411042ae6: Sending packet MESSAGE data 2["test",{"Message":"CREATE_BRIDGE_CONFIG"}]
241593151b6947759ed13264155849ba: Sending packet MESSAGE data 2["test",{"Message":"CREATE_BRIDGE_CONFIG"}]
8b89ee78af5d4c15a24e392a65034603: Sending packet MESSAGE data 2["test",{"Message":"CREATE_BRIDGE_CONFIG"}]
422a70e6720b4e28ba6abc23e5fc475c: Sending packet MESSAGE data 2["test",{"Message":"CREATE_BRIDGE_CONFIG"}]`

**Comment by miguelgrinberg** [comment]
I'm not familiar with the Vue-Socket.IO client and how it related to the official JS client, which is the one I use as a reference. The client is gone message indicates that the client isn't responding in a timely manner to the pings the server sends.

My guess is that you are using a version of server and client that are not compatible. The documentation has a version compatibility table that may be useful, but you'll need to find how to map it to your Vue client, because as I only track compatibility against the reference JS package.

**Comment by denisvitez** [comment]
That was my initial suspicion too, but I am not so sure about that anymore. I tried this sample client from python-socketio 
documentation:

`import socketio

sio = socketio.Client()
sio.connect('http://localhost:5000')
print('my sid is', sio.sid)`

But when I run it I get the following response:

`Traceback (most recent call last):
  File "/home/denis/code/socketio-client/main.py", line 22, in <module>
    sio.connect('http://localhost:5000')
  File "/home/denis/.virtualenvs/socketio-client/lib/python3.8/site-packages/socketio/client.py", line 284, in connect
    raise exceptions.ConnectionError(exc.args[0]) from None
socketio.exceptions.ConnectionError: Unexpected response from server`

All that gets printed in the debug on my server is:
`7933e122b58740a4a42260331b46923c: Sending packet OPEN data {'sid': '7933e122b58740a4a42260331b46923c', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
7933e122b58740a4a42260331b46923c: Sending packet MESSAGE data 0
127.0.0.1 - - [20/Jan/2021 19:15:11] "GET /socket.io/?transport=polling&EIO=4&t=1611166511.3898458 HTTP/1.1" 200 -
2f2c7daf114a442a95a4f7cb7de9a55e: Client is gone, closing socket
2f2c7daf114a442a95a4f7cb7de9a55e: Client is gone, closing socket
7933e122b58740a4a42260331b46923c: Client is gone, closing socket
7933e122b58740a4a42260331b46923c: Client is gone, closing socket`

So it seems like the client tries to do some communication, but fails. Like with the Vue.js client.

I am using the following packages:
`Flask-SocketIO==5.0.1 (on the Flask server)`
and
`python-socketio==5.0.4 (sample client)`

So based on the table in documentation they should get along (5.x) :)

**Comment by miguelgrinberg** [comment]
Can you also report on the versions of python-socketio and python-engineio on the server? And the python-engineio version used by your client?

**Comment by denisvitez** [comment]
I tried to update some of my packages and I noticed that there were incorrect versions of `python-engineio` and `python-socketio `installed in my Flask server. I corrected this and it seems to be working now :)

Now I have installed:
`python-engineio==4.0.0
python-socketio==5.0.4
Flask-SocketIO==5.0.1`

Should this get updated automatically when I updated the Flask-SocketIO and this was some strange pip behaviour, or do I need to be careful with that in the future and double check the versions? (I don't have much real experience with python).

Anyway thank you for your help and for providing this libraries in the first place :+1:
